src/hooks/useOrders.ts

```diff
@@ -81,19 +58,13 @@ export function useOrders(): Result {
         // check cancelled bool from parent scope
         if (cancelled) return
 
-        // Apply filters (remove deleted orders)
-        const filteredOrders = filterDeletedOrders(orders)
-
         // ensures we don't have multiple reruns for each update
         // i.e. offset change -> render
         //      isLoading change -> another render
         unstable_batchedUpdates(() => {
-          if (offset === 0) {
-            // fresh start/refresh: replace whatever is stored
-            dispatch(overwriteOrders(filteredOrders))
```

Comment:
Shouldn't we still overwrite orders when `offset == 0`. I don't think `updateOrders` quite replaces that

Reply:
Not really. The point of this change is exactly that. To update the list of orders 'in place' without flashing the UI.

`updateOrders` updates existing, remove deleted and appends new orders. That should cover all cases during an update, no?

Reply:
You are right. On second thought, shouldn't updateOrders be called unconditionally, without if orders.length
Because orders can be deleted all at once from the contact, if they were all previously cancelled

Reply:
> Because orders can be deleted all at once from the contact, if they were all previously cancelled

If orders are deleted from the contract, even if in a batch, they will still be returned but with all fields zeroed out.

That means, if orders.length === 0, there is really nothing to update.

Reply:
Cool. I need to reread the contract